build.gradle

```diff
@@ -45,7 +45,11 @@ subprojects {
     apply plugin: 'com.palantir.baseline-class-uniqueness'
 
     tasks.withType(JavaCompile) {
-        options.compilerArgs += ['-Werror']
+        options.compilerArgs += [
+            '-Werror',
+            '-Xlint:deprecation',
+            '-Xlint:unchecked',
+        ]
```

Comment:
Slight preference to do this sort of cleanup in a separate PR -- this way if something goes sideways and we revert, we don't lose unrelated cleanups! (no need to change this PR since it's already up, just thought I'd mention for next time)

Reply:
sorry, I snuck these deprecation and IDE warning fixes in as I was checking all the classes Gradle 8 compatibility.